fhir-codegen/gradle-plugin/src/main/kotlin/com/google/fhir/codegen/SerializerTypeSpecGenerator.kt

```diff
@@ -49,7 +49,14 @@ object SerializerTypeSpecGenerator {
   /** @param className the class the serializer is for */
   fun generate(className: ClassName): TypeSpec =
     TypeSpec.classBuilder(className.toSerializerClassName())
-      .addModifiers(KModifier.INTERNAL)
+      .apply {
+        // Mark serializers for backbone elements (e.g. Patient.contact) internal whilst keeping
+        // serializers for resources (e.g. Patient) public, with the assumption that class names
+        // with more than one single name are backbone elements.
+        if (className.simpleNames.size > 1) {
+          addModifiers(KModifier.INTERNAL)
```

Comment:
If the serializers for the nested class are exposed only internally, even though the outer class is public, don't you think, the serializations for the nested classes might be affected too outside the module?

Reply:
actually let me test this quickly

Reply:
:+1: